deploy.sh

```diff
@@ -153,16 +168,23 @@ function start_link_checker_processes {
         log "Rabbit is unavailable - sleeping"
         sleep 10
     done
-
-    docker-compose exec -T rabbitmq rabbitmqctl add_user di_website $RABBITMQ_PASSWORD
-    docker-compose exec -T rabbitmq rabbitmqctl add_vhost myvhost
-    docker-compose exec -T rabbitmq rabbitmqctl set_user_tags di_website di_website
-    docker-compose exec -T rabbitmq rabbitmqctl set_permissions -p myvhost di_website ".*" ".*" ".*"
+    if  docker-compose exec -T rabbitmq rabbitmqctl list_users | grep -q "root"; then
+        log "user already exists. Skipping ..."
+        docker-compose exec -T rabbitmq rabbitmqctl add_vhost myvhost
+        docker-compose exec -T rabbitmq rabbitmqctl set_user_tags root root
+        docker-compose exec -T rabbitmq rabbitmqctl set_permissions -p myvhost root ".*" ".*" ".*"
+    else
+        docker-compose exec -T rabbitmq rabbitmqctl add_user root $RABBITMQ_PASSWORD
+        docker-compose exec -T rabbitmq rabbitmqctl add_vhost myvhost
+        docker-compose exec -T rabbitmq rabbitmqctl set_user_tags root root
+        docker-compose exec -T rabbitmq rabbitmqctl set_permissions -p myvhost root ".*" ".*" ".*"
+    fi
```

Comment:
Not sure if this works anything like normal code... if so, can't you extract the common commands and only run the required ones within the `if` statement? i.e.
```
    if  docker-compose exec -T rabbitmq rabbitmqctl list_users | grep -q "root"; then
        log "user already exists. Skipping ..."
    else
        docker-compose exec -T rabbitmq rabbitmqctl add_user root $RABBITMQ_PASSWORD
    fi

    docker-compose exec -T rabbitmq rabbitmqctl add_vhost myvhost
    docker-compose exec -T rabbitmq rabbitmqctl set_user_tags root root
    docker-compose exec -T rabbitmq rabbitmqctl set_permissions -p myvhost root ".*" ".*" ".*"
```